From 571b407460a7c0f6c2d90835b12ec6c50845fd79 Mon Sep 17 00:00:00 2001
From: Steven Moreland <smoreland@google.com>
Date: Fri, 3 Sep 2021 22:39:10 +0000
Subject: [PATCH] libbinder: uptimeMillis returns int64_t! am: 3ba4963f5b am:
 17aa765fd3 am: d666af6990

Original change: https://googleplex-android-review.googlesource.com/c/platform/frameworks/native/+/15720346

Bug: 197336441

Change-Id: I703760ecbfa007d27caf76556ed21bbf0558df80
(cherry picked from commit 62eaabc3242b5e4ae5da7d39928123fda570f0b2)
---
 libs/binder/IServiceManager.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libs/binder/IServiceManager.cpp b/libs/binder/IServiceManager.cpp
index a9f2d7395..218970a1b 100644
--- a/libs/binder/IServiceManager.cpp
+++ b/libs/binder/IServiceManager.cpp
@@ -215,7 +215,8 @@ sp<IBinder> ServiceManagerShim::getService(const String16& name) const
 
     const bool isVendorService =
         strcmp(ProcessState::self()->getDriverName().c_str(), "/dev/vndbinder") == 0;
-    const long timeout = uptimeMillis() + 5000;
+    const long timeout = 5000;
+    int64_t startTime = uptimeMillis();
     // Vendor code can't access system properties
     if (!gSystemBootCompleted && !isVendorService) {
 #ifdef __ANDROID__
@@ -230,7 +231,7 @@ sp<IBinder> ServiceManagerShim::getService(const String16& name) const
     const long sleepTime = gSystemBootCompleted ? 1000 : 100;
 
     int n = 0;
-    while (uptimeMillis() < timeout) {
+    while (uptimeMillis() - startTime < timeout) {
         n++;
         ALOGI("Waiting for service '%s' on '%s'...", String8(name).string(),
             ProcessState::self()->getDriverName().c_str());
-- 
2.34.0.rc2.393.gf8c9666880-goog

